Firestore/Source/Local/FSTMemoryPersistence.mm

```diff
@@ -236,9 +237,14 @@ - (int)removeTargetsThroughSequenceNumber:(ListenSequenceNumber)sequenceNumber
 }
 
 - (int)removeOrphanedDocumentsThroughSequenceNumber:(ListenSequenceNumber)upperBound {
-  return [(FSTMemoryRemoteDocumentCache *)_persistence.remoteDocumentCache
-      removeOrphanedDocuments:self
-        throughSequenceNumber:upperBound];
+  std::vector<DocumentKey> removed =
+      [(FSTMemoryRemoteDocumentCache *)_persistence.remoteDocumentCache
+          removeOrphanedDocuments:self
+            throughSequenceNumber:upperBound];
+  for (auto it = removed.begin(); it != removed.end(); it++) {
+    _sequenceNumbers.erase(*it);
```

Comment:
I can't really tell what what `_sequenceNumbers` is tracking and why this important.

Could you add a comment to this member describing what it's tracking?